src/compiler/transformers/ts.ts

```diff
@@ -1889,7 +1889,23 @@ export function transformTypeScript(context: TransformationContext) {
     function transformEnumMemberDeclarationValue(member: EnumMember): Expression {
         const value = resolver.getConstantValue(member);
         if (value !== undefined) {
-            return typeof value === "string" ? factory.createStringLiteral(value) : factory.createNumericLiteral(value);
+            if (typeof value === "string") {
+                return factory.createStringLiteral(value);
+            }
+            if (Number.isNaN(value)) {

```

Comment:
alternatively, I could just not use the constant value in those cases and fallback to the other branch - that would likely work. But I feel that `NaN` and `Infinity` should not be passed to `factory.createNumericLiteral` so some changes to this branch of code here would have to be done anyway. I also like that those evaluated, yet shadowed, `Infinity` and `NaN` are normalized~ by the proposed changes.